muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application/current-student/workspaces.tsx

```diff
@@ -7,32 +7,37 @@ import { GuiderCurrentStudentStateType, GuiderStudentUserProfileType, GuiderType
 import {StateType} from '~/reducers';
 
 import '~/sass/elements/application-list.scss';
+import '~/sass/elements/empty.scss';
 
 interface CurrentStudentWorkspacesProps {
   i18n: i18nType,
-  guider: GuiderType
+  guider: GuiderType,
+  locale: string
 }
 
 interface CurrentStudentWorkspacesState {
 }
 
 class CurrentStudentWorkspaces extends React.Component<CurrentStudentWorkspacesProps, CurrentStudentWorkspacesState> {
   render(){
-    //TODO WAT?... mf???? o.o communicator?...
+    let reactComponent = this;
```

Comment:
Good, but this is anti pattern, use an arrow function in function (a, b) => and use `this` inside there, arrow functions don't lose context.... you can use `this` inside an arrow function and it will mantain the context of the parent.

Reply:
Something like `(a, b) => .... this.props.locale ......`